src/mesh.cpp

```diff
@@ -94,26 +94,45 @@ scene::IAnimatedMesh* createCubeMesh(v3f scale)
 
 void scaleMesh(scene::IMesh *mesh, v3f scale)
 {
-	if(mesh == NULL)
+	if (mesh == NULL)
 		return;
 
 	core::aabbox3d<f32> bbox;
-	bbox.reset(0,0,0);
+	bbox.reset(0, 0, 0);
 
 	u16 mc = mesh->getMeshBufferCount();
-	for(u16 j=0; j<mc; j++)
-	{
+	for (u16 j = 0; j < mc; j++) {
 		scene::IMeshBuffer *buf = mesh->getMeshBuffer(j);
-		video::S3DVertex *vertices = (video::S3DVertex*)buf->getVertices();
-		u16 vc = buf->getVertexCount();
-		for(u16 i=0; i<vc; i++)
-		{
-			vertices[i].Pos *= scale;
+		u16 vertex_count = buf->getVertexCount();
+		const video::E_VERTEX_TYPE vType = buf->getVertexType();
+		switch (vType) {
+			case video::EVT_STANDARD: {
```

Comment:
Again, this switch statement isn't even necessary.  Both video::S3DVertex2TCoords and video::S3DVertexTangents are instances of video::S3DVertex.  Why don't you change this?